deeppavlov/models/torch_bert/torch_transformers_el_ranker.py

```diff
@@ -0,0 +1,353 @@
+            threshold: Optional[float] = None,
+            **kwargs
+    ):
+        self.encoder_save_path = encoder_save_path
+        self.bilinear_save_path = bilinear_save_path
+        self.pretrained_bert = pretrained_bert
+        self.bert_config_file = bert_config_file
+        self.return_probas = return_probas
+        self.attention_probs_keep_prob = attention_probs_keep_prob
+        self.hidden_keep_prob = hidden_keep_prob
+        self.clip_norm = clip_norm
+
+        super().__init__(
+            model_name=model_name,
+            optimizer=optimizer,
+            criterion=criterion,
+            optimizer_parameters=optimizer_parameters,
+            return_probas=return_probas,
+            **kwargs)
+
+    def train_on_batch(self, q_features: List[Dict],
+                       c_features: List[Dict],
+                       entity_tokens_pos: List[int],
+                       labels: List[int]) -> float:
+
+        _input = {'labels': labels}
+        _input['entity_tokens_pos'] = entity_tokens_pos
+        for elem in ['input_ids', 'attention_mask']:
+            inp_elem = [getattr(f, elem) for f in q_features]
+            _input[f"q_{elem}"] = torch.LongTensor(inp_elem).to(self.device)
+        for elem in ['input_ids', 'attention_mask']:
+            inp_elem = [getattr(f, elem) for f in c_features]
+            _input[f"c_{elem}"] = torch.LongTensor(inp_elem).to(self.device)
+
+        self.model.train()
+        self.model.zero_grad()
+        self.optimizer.zero_grad()  # zero the parameter gradients
+
+        loss, softmax_scores = self.model(**_input)
+        loss.backward()
+        self.optimizer.step()
+
+        # Clip the norm of the gradients to prevent the "exploding gradients" problem
+        if self.clip_norm:
+            torch.nn.utils.clip_grad_norm_(self.model.parameters(), self.clip_norm)
+
+        if self.lr_scheduler is not None:
+            self.lr_scheduler.step()
+
+        return loss.item()
+
+    def __call__(self, q_features: List[Dict],
+                 c_features: List[Dict],
+                 entity_tokens_pos: List[int]) -> Union[List[int], List[np.ndarray]]:
+
+        self.model.eval()
+
+        _input = {'entity_tokens_pos': entity_tokens_pos}
+        for elem in ['input_ids', 'attention_mask']:
+            inp_elem = [getattr(f, elem) for f in q_features]
+            _input[f"q_{elem}"] = torch.LongTensor(inp_elem).to(self.device)
+        for elem in ['input_ids', 'attention_mask']:
+            inp_elem = [getattr(f, elem) for f in c_features]
+            _input[f"c_{elem}"] = torch.LongTensor(inp_elem).to(self.device)
+
+        with torch.no_grad():
+            softmax_scores = self.model(**_input)
+            if self.return_probas:
+                pred = softmax_scores
+            else:
+                pred = torch.argmax(softmax_scores, dim=1).cpu().numpy()
+
+        return pred
+
+    def siamese_ranking_el_model(self, **kwargs) -> nn.Module:
```

Comment:
why you need kwargs? And why object initialization is standalone method?

Reply:
I used the example from https://github.com/deepmipt/DeepPavlov/blob/191e3603cd06267a877cd711e3ffc11232524496/deeppavlov/models/classifiers/torch_classification_model.py#L182